### PR TITLE
added create_uart to wrap busio.UART in a ManagedDevice

### PR DESCRIPTION
**Pull request by anniexiang01**

### CODE REVIEW

File: shared/lib/pin_manager.py [hunk 1]
```diff
@@ -216,6 +216,18 @@ def create_i2c(self, scl, sda, frequency=100000):
...
 
+    def create_uart(self, tx, rx, baudrate=50000):
+        """
+        Creates and returns ManagedDevice wrapping a busio.UART on the two specified pins
+        with the specified baudrate, or returns one from the cache if one has already
+        been created.
+        """
+        return self._create_general_device(
+            [tx, rx],
+            (busio.UART, baudrate),
```
Comment (Randall-Scharpf) [inline shared/lib/pin_manager.py]: The `baudrate` parameter should not be used as a part of the key that identifies a `busio.UART` as a particular hardware device since the library allows `baudrate` to be reconfigure.

File: shared/lib/pin_manager.py [hunk 2]
```diff
@@ -216,6 +216,18 @@ def create_i2c(self, scl, sda, frequency=100000):
...
+            (lambda: busio.UART(tx, rx, baudrate=baudrate)),
```
Comment (Randall-Scharpf) [inline shared/lib/pin_manager.py]: There's no need to explicitly set the baudrate in the constructor for a `busio.UART`, since it can be reset through the `baudrate` property at a later time. It's alright to include it, but it's just not needed.

It would make more sense to include the `bits`, `parity`, and `stop` parameters since some devices require different values for them and they can't be reconfigured later. I don't think we're interested in hardware RTS/CTS for RS232 or hardware TE for RS485, so those parameters are less necessary, but again they can't be reconfigured after construction, so they're candidates for inclusion here.

**Review by Randall-Scharpf — CHANGES_REQUESTED** [review]
I added a few questions about how and why you're handling optional arguments in the way you are. Looks like the code should work in basic cases and is almost right, just a bit of tweaking to future-proof it for edge cases.

File: config/pylama.cfg [hunk 3]
```diff
@@ -16,7 +16,7 @@ linters = pylint,pyflakes,radon
 # Note: the following are added to fprime's list of ignored tests for RAPID-0 because of the reasons stated
 ## E0401 is for imports, which we check using our unit testing workflow in order to correctly handle CircuitPython-specific modules
 ## R0903 requires at least two public methods per class, which is not a RAPID-0 style requirement
-disable = C0301,W0612,E0401,R0903
+disable = C0301,W0612,E0401,R0903,R0913
```
Comment (Randall-Scharpf) [inline config/pylama.cfg:20]: Can you add a comment indicating what R0913 is so that we remember in the future why it's disabled?

**Review by Randall-Scharpf — CHANGES_REQUESTED** [review]

**Review by Randall-Scharpf — APPROVED** [review]
